packages/EasyDecision/src/Configurators/AddRulesDecisionConfigurator.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EonX\EasyDecision\Configurators;
+
+use EonX\EasyDecision\Interfaces\DecisionInterface;
+use EonX\EasyDecision\Interfaces\RestrictedRuleInterface;
+
+final class AddRulesDecisionConfigurator extends AbstractConfigurator
+{
+    /**
+     * @var iterable<\EonX\EasyDecision\Interfaces\RuleInterface>
+     */
+    private $rules;
+
+    /**
+     * @param iterable<\EonX\EasyDecision\Interfaces\RuleInterface> $rules
+     */
+    public function __construct(iterable $rules, ?int $priority = null)
+    {
+        parent::__construct($priority);
+
+        $this->rules = $rules;
```

Comment:
I would like to add some logic to make sure we have only rules in this, have a look at other packages how we filter the iterables